src/common/transformations/src/transformations/common_optimizations/remove_multi_subgraph_op_dangling_params.cpp

```diff
@@ -97,8 +146,47 @@ ov::pass::RemoveMultiSubGraphOpDanglingParams::RemoveMultiSubGraphOpDanglingPara
             }
             multi_subgraph_op->set_arguments(op_inputs);
         }
-        return false;
-    };
-    auto m = std::make_shared<pattern::Matcher>(multi_subgraph_op_pattern, matcher_name);
-    this->register_matcher(m, callback);
+        if (!outputs_to_remove.empty()) {
+            // we need to reconstruct operation with new number of outputs, we cannot reduce number of outputs of
+            // existing op
+            std::shared_ptr<ov::op::util::MultiSubGraphOp> new_op;
+            if (if_op) {
+                new_op = std::make_shared<opset10::If>(if_op->input_value(0));
+                new_op->set_arguments(multi_subgraph_op->input_values());
+            } else if (loop_op) {
+                auto new_loop_op = std::make_shared<opset10::Loop>(loop_op->input_value(0), loop_op->input_value(1));
+                new_loop_op->set_special_body_ports(loop_op->get_special_body_ports());
```

Comment:
are you sure if the special port won't change the id after removing outputs?

Reply:
Agree, need to ensure that it gets changed if necessary